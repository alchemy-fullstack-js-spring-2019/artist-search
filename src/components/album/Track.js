import React from 'react';
import PropTypes from 'prop-types';

export default function Track({ artist, songTitle, lyricsLoad }) {
  return <a href='javascript:void(0)' onClick={() => {
    return lyricsLoad(artist, songTitle);
  }}>{songTitle}</a>;
}

Track.propTypes = {
  artist: PropTypes.string.isRequired,
  songTitle: PropTypes.string.isRequired,
  lyricsLoad: PropTypes.func.isRequired
};
