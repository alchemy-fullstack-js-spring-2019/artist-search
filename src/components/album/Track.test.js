import React from 'react';
import { shallow } from 'enzyme';
import Track from './Track';

describe('Track component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Track artist='' songTitle='' lyricsLoad={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
