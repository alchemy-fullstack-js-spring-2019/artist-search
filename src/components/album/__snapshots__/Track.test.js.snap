// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track component matches snapshot 1`] = `
<a
  href="javascript:void(0)"
  onClick={[Function]}
/>
`;
